homework/index.js

```diff
@@ -16,31 +30,97 @@
     xhr.send();
   }
 
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach(key => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.textContent = value;
-      } else {
-        elem.setAttribute(key, value);
-      }
+  function renderRepositories(repo, ulRepoInfo) {
+    const url = `https://api.github.com/repos/HackYourFuture/${repo}`;
+    fetchJSON(url, (err, repoInfo) => {
+      ulRepoInfo.innerHTML = '';
+      const li = createAndAppend('li', ulRepoInfo, { id: 'li1' });
+      const table = createAndAppend('table', li);
+      const tbody = createAndAppend('tbody', table);
+
+      const tr1 = createAndAppend('tr', tbody);
+      createAndAppend('td', tr1, { text: 'Repository:', class: 'label' });
+      createAndAppend('td', tr1);
+      createAndAppend('a', tr1.lastChild, {
+        href: repoInfo.html_url,
+        target: '_blank',
+        text: repoInfo.name,
+      });
+
+      const tr2 = createAndAppend('tr', tbody);
+      createAndAppend('td', tr2, { text: 'Description:', class: 'label' });
+      createAndAppend('td', tr2, { text: `${repoInfo.description}` });
+
+      const tr3 = createAndAppend('tr', tbody);
+      createAndAppend('td', tr3, { text: 'Forks:', class: 'label' });
+      createAndAppend('td', tr3, { text: `${repoInfo.forks}` });
+
+      const tr4 = createAndAppend('tr', tbody);
+      createAndAppend('td', tr4, { text: 'Updated:', class: 'label' });
+      createAndAppend('td', tr4, { text: new Date(repoInfo.updated_at).toLocaleString() });
     });
-    return elem;
+  }
+
+  function renderContributors(repoName, ulContributorInfo) {
+    const url = `https://api.github.com/repos/HackYourFuture/${repoName}/contributors`;
+    fetchJSON(url, (err, contributors) => {
+      ulContributorInfo.innerHTML = '';
+      contributors.forEach(contributor => {
+        const listItems = createAndAppend('li', ulContributorInfo, { class: 'li2' });
```

Comment:
1. You are creating a single list item at a time. So the singular form `listItem` should be used for this variable.
2. Avoid (CSS or otherwise) names with numeric suffixes. Take a little more time to come up with a good name.
3. It would be nice if you could click anywhere on the list item to go to the contributors page. You can achieve that by wrapping all inner content of each list item into an `a` tag with the relevant url for the `href`. This, in place of the `a` you have used for the contributor login name.

Reply:
1. Yes single item at a time. But some repositories have more than one contributor. So should it be plural?
3. I guess I misunderstood hw. Yes I am gonna correct it.